pkg/providers/vault/vault.go

```diff
@@ -176,6 +178,9 @@ func (p *provider) ensureClient() (*vault.Client, error) {
 			p.debugf("Vault connections failed")
 			return nil, fmt.Errorf("Cannot create Vault Client: %v", err)
 		}
+		if p.Namespace != "" {
+			cli.setNamespace(p.Namespace)
```

Comment:
Hey! Thanks for contributing.

Any struct member that starts with a lower-case character is private in Go so this would result in a compile error.
Could you refer to the documentation of the Vault client for the correct function?

Reply:
Doh! I did actually look at the docs, but clearly camel-case runs strong in my blood or something.

https://pkg.go.dev/github.com/hashicorp/vault/api#Client.SetNamespace

This is fixed now.